{"name": "test_community_redirection", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n\tGetHandleVerifier [0x00BAC7F3+24435]\n\t(No symbol) [0x00B32074]\n\t(No symbol) [0x00A006E3]\n\t(No symbol) [0x00A48B39]\n\t(No symbol) [0x00A48E8B]\n\t(No symbol) [0x00A91AC2]\n\t(No symbol) [0x00A6D804]\n\t(No symbol) [0x00A8F20A]\n\t(No symbol) [0x00A6D5B6]\n\t(No symbol) [0x00A3C54F]\n\t(No symbol) [0x00A3D894]\n\tGetHandleVerifier [0x00EB70A3+3213347]\n\tGetHandleVerifier [0x00ECB0C9+3295305]\n\tGetHandleVerifier [0x00EC558C+3271948]\n\tGetHandleVerifier [0x00C47360+658144]\n\t(No symbol) [0x00B3B27D]\n\t(No symbol) [0x00B38208]\n\t(No symbol) [0x00B383A9]\n\t(No symbol) [0x00B2AAC0]\n\tBaseThreadInitThunk [0x76595D49+25]\n\tRtlInitializeExceptionChain [0x770ACE3B+107]\n\tRtlGetAppContainerNamedObjectPath [0x770ACDC1+561]", "trace": "self = <urllib3.connectionpool.HTTPConnectionPool object at 0x000001E55DB57770>, conn = <urllib3.connection.HTTPConnection object at 0x000001E55DB57CB0>, method = 'POST'\nurl = '/session/b502842614286347a3f1ef5cb3997bb3/execute/sync'\nbody = '{\"script\": \"arguments[0].click();\", \"args\": [{\"element-6066-11e4-a52e-4f735466cecf\": \"f.3D6E01CBFA658E3DFBF682E270395FF7.d.D0968FD2056A01F67BF1DF7A45410085.e.58\"}]}'\nheaders = HTTPHeaderDict({'Accept': 'application/json', 'Content-Type': 'application/json;charset=UTF-8', 'User-Agent': 'selenium/4.29.0 (python windows)', 'Connection': 'keep-alive'})\nretries = Retry(total=3, connect=None, read=None, redirect=None, status=None), timeout = Timeout(connect=120, read=120, total=None), chunked = False, response_conn = None, preload_content = True, decode_content = True\nenforce_content_length = True\n\n    def _make_request(\n        self,\n        conn: BaseHTTPConnection,\n        method: str,\n        url: str,\n        body: _TYPE_BODY | None = None,\n        headers: typing.Mapping[str, str] | None = None,\n        retries: Retry | None = None,\n        timeout: _TYPE_TIMEOUT = _DEFAULT_TIMEOUT,\n        chunked: bool = False,\n        response_conn: BaseHTTPConnection | None = None,\n        preload_content: bool = True,\n        decode_content: bool = True,\n        enforce_content_length: bool = True,\n    ) -> BaseHTTPResponse:\n        \"\"\"\n        Perform a request on a given urllib connection object taken from our\n        pool.\n    \n        :param conn:\n            a connection from one of our connection pools\n    \n        :param method:\n            HTTP request method (such as GET, POST, PUT, etc.)\n    \n        :param url:\n            The URL to perform the request on.\n    \n        :param body:\n            Data to send in the request body, either :class:`str`, :class:`bytes`,\n            an iterable of :class:`str`/:class:`bytes`, or a file-like object.\n    \n        :param headers:\n            Dictionary of custom headers to send, such as User-Agent,\n            If-None-Match, etc. If None, pool headers are used. If provided,\n            these headers completely replace any pool-specific headers.\n    \n        :param retries:\n            Configure the number of retries to allow before raising a\n            :class:`~urllib3.exceptions.MaxRetryError` exception.\n    \n            Pass ``None`` to retry until you receive a response. Pass a\n            :class:`~urllib3.util.retry.Retry` object for fine-grained control\n            over different types of retries.\n            Pass an integer number to retry connection errors that many times,\n            but no other types of errors. Pass zero to never retry.\n    \n            If ``False``, then retries are disabled and any exception is raised\n            immediately. Also, instead of raising a MaxRetryError on redirects,\n            the redirect response will be returned.\n    \n        :type retries: :class:`~urllib3.util.retry.Retry`, False, or an int.\n    \n        :param timeout:\n            If specified, overrides the default timeout for this one\n            request. It may be a float (in seconds) or an instance of\n            :class:`urllib3.util.Timeout`.\n    \n        :param chunked:\n            If True, urllib3 will send the body using chunked transfer\n            encoding. Otherwise, urllib3 will send the body using the standard\n            content-length form. Defaults to False.\n    \n        :param response_conn:\n            Set this to ``None`` if you will handle releasing the connection or\n            set the connection to have the response release it.\n    \n        :param preload_content:\n          If True, the response's body will be preloaded during construction.\n    \n        :param decode_content:\n            If True, will attempt to decode the body based on the\n            'content-encoding' header.\n    \n        :param enforce_content_length:\n            Enforce content length checking. Body returned by server must match\n            value of Content-Length header, if present. Otherwise, raise error.\n        \"\"\"\n        self.num_requests += 1\n    \n        timeout_obj = self._get_timeout(timeout)\n        timeout_obj.start_connect()\n        conn.timeout = Timeout.resolve_default_timeout(timeout_obj.connect_timeout)\n    \n        try:\n            # Trigger any extra validation we need to do.\n            try:\n                self._validate_conn(conn)\n            except (SocketTimeout, BaseSSLError) as e:\n                self._raise_timeout(err=e, url=url, timeout_value=conn.timeout)\n                raise\n    \n        # _validate_conn() starts the connection to an HTTPS proxy\n        # so we need to wrap errors with 'ProxyError' here too.\n        except (\n            OSError,\n            NewConnectionError,\n            TimeoutError,\n            BaseSSLError,\n            CertificateError,\n            SSLError,\n        ) as e:\n            new_e: Exception = e\n            if isinstance(e, (BaseSSLError, CertificateError)):\n                new_e = SSLError(e)\n            # If the connection didn't successfully connect to it's proxy\n            # then there\n            if isinstance(\n                new_e, (OSError, NewConnectionError, TimeoutError, SSLError)\n            ) and (conn and conn.proxy and not conn.has_connected_to_proxy):\n                new_e = _wrap_proxy_error(new_e, conn.proxy.scheme)\n            raise new_e\n    \n        # conn.request() calls http.client.*.request, not the method in\n        # urllib3.request. It also calls makefile (recv) on the socket.\n        try:\n            conn.request(\n                method,\n                url,\n                body=body,\n                headers=headers,\n                chunked=chunked,\n                preload_content=preload_content,\n                decode_content=decode_content,\n                enforce_content_length=enforce_content_length,\n            )\n    \n        # We are swallowing BrokenPipeError (errno.EPIPE) since the server is\n        # legitimately able to close the connection after sending a valid response.\n        # With this behaviour, the received response is still readable.\n        except BrokenPipeError:\n            pass\n        except OSError as e:\n            # MacOS/Linux\n            # EPROTOTYPE and ECONNRESET are needed on macOS\n            # https://erickt.github.io/blog/2014/11/19/adventures-in-debugging-a-potential-osx-kernel-bug/\n            # Condition changed later to emit ECONNRESET instead of only EPROTOTYPE.\n            if e.errno != errno.EPROTOTYPE and e.errno != errno.ECONNRESET:\n                raise\n    \n        # Reset the timeout for the recv() on the socket\n        read_timeout = timeout_obj.read_timeout\n    \n        if not conn.is_closed:\n            # In Python 3 socket.py will catch EAGAIN and return None when you\n            # try and read into the file pointer created by http.client, which\n            # instead raises a BadStatusLine exception. Instead of catching\n            # the exception and assuming all BadStatusLine exceptions are read\n            # timeouts, check for a zero timeout before making the request.\n            if read_timeout == 0:\n                raise ReadTimeoutError(\n                    self, url, f\"Read timed out. (read timeout={read_timeout})\"\n                )\n            conn.timeout = read_timeout\n    \n        # Receive the response from the server\n        try:\n>           response = conn.getresponse()\n\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:534: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\connection.py:516: in getresponse\n    httplib_response = super().getresponse()\nC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.13_3.13.752.0_x64__qbz5n2kfra8p0\\Lib\\http\\client.py:1430: in getresponse\n    response.begin()\nC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.13_3.13.752.0_x64__qbz5n2kfra8p0\\Lib\\http\\client.py:331: in begin\n    version, status, reason = self._read_status()\nC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.13_3.13.752.0_x64__qbz5n2kfra8p0\\Lib\\http\\client.py:292: in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <socket.SocketIO object at 0x000001E55DB6ADA0>, b = <memory at 0x000001E55DAD9900>\n\n    def readinto(self, b):\n        \"\"\"Read up to len(b) bytes into the writable buffer *b* and return\n        the number of bytes read.  If the socket is non-blocking and no bytes\n        are available, None is returned.\n    \n        If *b* is non-empty, a 0 return value indicates that the connection\n        was shutdown at the other end.\n        \"\"\"\n        self._checkClosed()\n        self._checkReadable()\n        if self._timeout_occurred:\n            raise OSError(\"cannot read from timed out object\")\n        try:\n>           return self._sock.recv_into(b)\nE           TimeoutError: timed out\n\nC:\\Program Files\\WindowsApps\\PythonSoftwareFoundation.Python.3.13_3.13.752.0_x64__qbz5n2kfra8p0\\Lib\\socket.py:719: TimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nself = <pageObjects.home_page.HomePage object at 0x000001E55DB56660>, text = 'River'\nsuggestions = [<selenium.webdriver.remote.webelement.WebElement (session=\"b502842614286347a3f1ef5cb3997bb3\", element=\"f.3D6E01CBFA65...14286347a3f1ef5cb3997bb3\", element=\"f.3D6E01CBFA658E3DFBF682E270395FF7.d.D0968FD2056A01F67BF1DF7A45410085.e.59\")>, ...]\nsearch_type = 'community'\n\n    def click_on_suggestion(self, text, suggestions, search_type):\n        chosenProductName = \"\"\n    \n        try:\n            # Try to execute the main logic\n            if not suggestions:\n                raise ValueError(\"No suggestions found. Retrying...\")  # Force retry if list is empty\n    \n            # Choose a random element from the suggestions\n            chosen_element = random.choice(suggestions)\n    \n            # Scroll the element into view\n            self.scroll_into_view(chosen_element)\n    \n            # Re-locate the element in case the reference is stale\n            suggestions = self.get_locator_value(search_type)  # Re-fetch the suggestions\n            chosen_element = random.choice(suggestions)  # Re-select the element\n            chosenProductName = chosen_element.get_attribute(\"aria-label\")\n    \n            # Wait until it's clickable\n            WebDriverWait(self.driver, 20).until(EC.element_to_be_clickable(chosen_element))\n    \n            # Forcefully click using JavaScript Executor\n>           self.driver.execute_script(\"arguments[0].click();\", chosen_element)\n\npageObjects\\home_page.py:74: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n..\\automation_framework\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:528: in execute_script\n    return self.execute(command, {\"script\": script, \"args\": converted_args})[\"value\"]\n..\\automation_framework\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:427: in execute\n    response = self.command_executor.execute(driver_command, params)\n..\\automation_framework\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\remote_connection.py:404: in execute\n    return self._request(command_info[0], url, body=data)\n..\\automation_framework\\.venv\\Lib\\site-packages\\selenium\\webdriver\\remote\\remote_connection.py:428: in _request\n    response = self._conn.request(method, url, body=body, headers=headers, timeout=self._client_config.timeout)\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\_request_methods.py:143: in request\n    return self.request_encode_body(\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\_request_methods.py:278: in request_encode_body\n    return self.urlopen(method, url, **extra_kw)\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\poolmanager.py:443: in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:841: in urlopen\n    retries = retries.increment(\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\util\\retry.py:474: in increment\n    raise reraise(type(error), error, _stacktrace)\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\util\\util.py:39: in reraise\n    raise value\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:787: in urlopen\n    response = self._make_request(\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:536: in _make_request\n    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <urllib3.connectionpool.HTTPConnectionPool object at 0x000001E55DB57770>, err = TimeoutError('timed out'), url = '/session/b502842614286347a3f1ef5cb3997bb3/execute/sync', timeout_value = 120\n\n    def _raise_timeout(\n        self,\n        err: BaseSSLError | OSError | SocketTimeout,\n        url: str,\n        timeout_value: _TYPE_TIMEOUT | None,\n    ) -> None:\n        \"\"\"Is the error actually a timeout? Will raise a ReadTimeout or pass\"\"\"\n    \n        if isinstance(err, SocketTimeout):\n>           raise ReadTimeoutError(\n                self, url, f\"Read timed out. (read timeout={timeout_value})\"\n            ) from err\nE           urllib3.exceptions.ReadTimeoutError: HTTPConnectionPool(host='localhost', port=63909): Read timed out. (read timeout=120)\n\n..\\automation_framework\\.venv\\Lib\\site-packages\\urllib3\\connectionpool.py:367: ReadTimeoutError\n\nDuring handling of the above exception, another exception occurred:\n\nself = <test_sorting.TestSorting object at 0x000001E55DB79F90>, driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"b502842614286347a3f1ef5cb3997bb3\")>\n\n    def test_community_redirection(self, driver):\n        home_page = HomePage(driver)\n        community_page = CommunityPage(driver)\n    \n>       chosenProductName = home_page.search(\"River\", \"community\")\n\ntests\\test_sorting.py:13: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npageObjects\\home_page.py:31: in search\n    return self.click_on_suggestion(text, suggestions, search_type)\npageObjects\\home_page.py:82: in click_on_suggestion\n    self.enter_text_one_by_one(HomePageLocators.SEARCH_BOX, text)\npageObjects\\base_page.py:25: in enter_text_one_by_one\n    element = self.wait.until(EC.presence_of_element_located(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"b502842614286347a3f1ef5cb3997bb3\")>, method = <function presence_of_element_located.<locals>._predicate at 0x000001E55DB9C7C0>, message = ''\n\n    def until(self, method: Callable[[D], Union[Literal[False], T]], message: str = \"\") -> T:\n        \"\"\"Wait until the method returns a value that is not False.\n    \n        Calls the method provided with the driver as an argument until the\n        return value does not evaluate to ``False``.\n    \n        Parameters:\n        ----------\n        method: callable(WebDriver)\n            - A callable object that takes a WebDriver instance as an argument.\n    \n        message: str\n            - Optional message for :exc:`TimeoutException`\n    \n        Return:\n        -------\n        object: T\n            - The result of the last call to `method`\n    \n        Raises:\n        -------\n        TimeoutException\n            - If 'method' does not return a truthy value within the WebDriverWait\n            object's timeout\n    \n        Example:\n        --------\n        >>> from selenium.webdriver.common.by import By\n        >>> from selenium.webdriver.support.ui import WebDriverWait\n        >>> from selenium.webdriver.support import expected_conditions as EC\n    \n        # Wait until an element is visible on the page\n        >>> wait = WebDriverWait(driver, 10)\n        >>> element = wait.until(EC.visibility_of_element_located((By.ID, \"exampleId\")))\n        >>> print(element.text)\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            if time.monotonic() > end_time:\n                break\n            time.sleep(self._poll)\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       \tGetHandleVerifier [0x00BAC7F3+24435]\nE       \t(No symbol) [0x00B32074]\nE       \t(No symbol) [0x00A006E3]\nE       \t(No symbol) [0x00A48B39]\nE       \t(No symbol) [0x00A48E8B]\nE       \t(No symbol) [0x00A91AC2]\nE       \t(No symbol) [0x00A6D804]\nE       \t(No symbol) [0x00A8F20A]\nE       \t(No symbol) [0x00A6D5B6]\nE       \t(No symbol) [0x00A3C54F]\nE       \t(No symbol) [0x00A3D894]\nE       \tGetHandleVerifier [0x00EB70A3+3213347]\nE       \tGetHandleVerifier [0x00ECB0C9+3295305]\nE       \tGetHandleVerifier [0x00EC558C+3271948]\nE       \tGetHandleVerifier [0x00C47360+658144]\nE       \t(No symbol) [0x00B3B27D]\nE       \t(No symbol) [0x00B38208]\nE       \t(No symbol) [0x00B383A9]\nE       \t(No symbol) [0x00B2AAC0]\nE       \tBaseThreadInitThunk [0x76595D49+25]\nE       \tRtlInitializeExceptionChain [0x770ACE3B+107]\nE       \tRtlGetAppContainerNamedObjectPath [0x770ACDC1+561]\n\n..\\automation_framework\\.venv\\Lib\\site-packages\\selenium\\webdriver\\support\\wait.py:146: TimeoutException"}, "attachments": [{"name": "stdout", "source": "009d8da3-5561-429b-9991-2f6fc1a39262-attachment.txt", "type": "text/plain"}], "start": 1743575286519, "stop": 1743575469354, "uuid": "b01207c1-5bfa-4d76-b34e-af3e3c1b188c", "historyId": "edfddfcafa34cdb70ba6f7d2e7e7a4db", "testCaseId": "edfddfcafa34cdb70ba6f7d2e7e7a4db", "fullName": "tests.test_sorting.TestSorting#test_community_redirection", "labels": [{"name": "tag", "value": "@pytest.mark.usefixtures('driver')"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_sorting"}, {"name": "subSuite", "value": "TestSorting"}, {"name": "host", "value": "EXIPC0261"}, {"name": "thread", "value": "25052-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_sorting"}]}